docs/guides/auth.md

```diff
@@ -22,38 +22,50 @@ authentication.
 
 ###OAuth 2.0
 
-gRPC provides a generic mechanism (described below) to attach metadata to requests
-and responses. This mechanism can be used to attach OAuth 2.0 Access Tokens to
-RPCs being made at a client. Additional support for acquiring Access Tokens while
+gRPC provides a generic mechanism (described below) to attach metadata based credentials
+to requests and responses. Additional support for acquiring Access Tokens while
 accessing Google APIs through gRPC is provided for certain auth flows, demonstrated
 through code examples below.
 
-*WARNING*: OAuth2 shall only be used to connect to Google services. Sending a
-Google issued OAuth2 token to a non-Google service could result in this token
+*WARNING*: Google OAuth2 credentials shall only be used to connect to Google services.
+Sending a Google issued OAuth2 token to a non-Google service could result in this token
 being stolen and used to impersonate the client to Google services.
 
 ## API
 
 To reduce complexity and minimize API clutter, gRPC works with a unified concept of
-a Credentials object. Users construct gRPC credentials using corresponding bootstrap
-credentials (e.g., SSL client certs or Service Account Keys), and use the
-credentials while creating a gRPC channel to any server. Depending on the type of
-credential supplied, the channel uses the credentials during the initial SSL/TLS
-handshake with the server, or uses  the credential to generate and attach Access
-Tokens to each request being made on the channel.
+Credentials objects.
+
+Credentials can be of two types:
+
+- Channel Credentials which are attached to a channel such as SSL credentials.
+- Call Credentials which are attached to a call (or `ClientContext` in C++).
+
+Credentials can be composed using `CompositeChannelCredentials` which associates
+a `ChannelCredentials` and a `CallCredentials` in order to create a new
+`ChannelCredentials`. The result will, for each call on the channel, send the
+authentication data associated with the composed `CallCredentials`.
+
+For example, a `ChannelCredentials` could be created from an `SslCredentials`
+and an `AccessTokenCredentials`. The result, applied to a `Channel` would send
```

Comment:
minor change: The result when applied to a `Channel` would send the appropriate access token for each call on this channel